File: eftl-dotnet-sdk/TIBCO.EFTL/websocket/WebSocketListener.cs
/*
 * Copyright (c) 2013-$Date: 2014-10-06 14:32:56 -0500 (Mon, 06 Oct 2014) $ TIBCO Software Inc.
 * Licensed under a BSD-style license. Refer to [LICENSE]
 * For more information, please contact:
 * TIBCO Software Inc., Palo Alto, California, USA
 *
 * $Id: WebSocketListener.cs 75907 2014-10-06 19:32:56Z bmahurka $
 *
 */
using System;
using System.Collections;
using System.Linq;
using System.Text;

namespace TIBCO.EFTL
{
    public interface WebSocketListener 
    {
        void OnOpen();
        
        void OnClose(int code, String reason);
    
        void OnError(Exception cause);
        
        void OnMessage(String text);
        
        void OnMessage(byte[] data, int offset, int length);
        
        void OnPong(byte[] data, int offset, int length);
    }
}

File: eftl-java-sdk/src/com/tibco/eftl/websocket/WebSocketListener.java
/*
 * Copyright (c) 2013-$Date: 2013-12-04 09:51:34 -0800 (Wed, 04 Dec 2013) $ TIBCO Software Inc.
 * Licensed under a BSD-style license. Refer to [LICENSE]
 * For more information, please contact:
 * TIBCO Software Inc., Palo Alto, California, USA
 *
 * $Id: WebSocketListener.java 70851 2013-12-04 17:51:34Z $
 *
 */
package com.tibco.eftl.websocket;

public interface WebSocketListener {

    public void onOpen();
    
    public void onClose(int code, String reason);
    
    public void onError(Throwable cause);

    public void onMessage(String text);
    
    public void onMessage(byte[] data, int offset, int length);
    
    public void onPong(byte[] data, int offset, int length);
}
